pydatastructs/linear_data_structures/stack.py

```diff
@@ -0,0 +1,120 @@
+
+from __future__ import print_function, division
+from copy import deepcopy as dc
+
+__all__ = [
+    'Stack'
+]
+
+_clean = lambda x: x\
+    .replace("<class '", '', 1)\
+    .replace("<type '", '', 1)\
+    .replace("'>", '', 1)
+_check_type = lambda a, t: isinstance(a, t)
+NoneType = type(None)
+
+class Stack(object):
+    """Class defination for 'Stack' datatype
+    
+    Parameters
+    ----------
+    max_size : int, optional
+        Maximum size of stack allowed
+    type_restriction : list, optional
+        List of types(as strings) for element which can be inserted into Stack, provide empty list for no restrictions.
+
+    Raises
+    ------
+    TypeError
+        max_size argument should of type 'int'.
+    TypeError
+        type_restriction argument takes 'list' type only.
+    TypeError
+        All types in type_restriction list should be provided in string format
+
+    Example
+    -------
+    >>> from pydatastructs import Stack
+    >>> my_stack = Stack()
+    >>> my_stack.push(1)
+    >>> my_stack.push(2)
+    >>> my_stack.pop()
+    2
+    >>> print(my_stack)
+    <Stack length:[1]>
+    """
+
+    def __init__(
+        self, max_size=10 ** 15, type_restriction=list()
```

Comment:
Let's keep the max size unbounded. We can keep it dependent on the amount of memory available.